core/templates/domain/objects/objects-domain.constants.spec.ts

```diff
@@ -0,0 +1,137 @@
+// Copyright 2024 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview unit tests for the units object domain constants.
+ */
+
+import {ObjectsDomainConstants} from './objects-domain.constants';
+import {getCurrencyUnits} from './NumberWithUnitsObjectFactory';
+import {all, create} from 'mathjs';
+
+const math = create(all);
+
+const getUnitPrefixes = (): string[] => {
+  const prefixes = math.Unit.PREFIXES;
+  let prefixSet = new Set<string>();
+  for (const name in prefixes) {
+    if (name === 'NONE') {
+      continue;
+    }
+    for (const prefix in prefixes[name]) {
+      if (prefix === '') {
+        continue;
+      }
+      prefixSet.add(prefix);
+    }
+  }
+
+  return Array.from(prefixSet);
+};
+
+const getAllMathjsUnits = (): string[] => {
+  let mathjsUnits: string[] = [];
+  for (const unit in math.Unit.UNITS) {
+    mathjsUnits.push(unit);
+  }
+  return mathjsUnits;
+};
+
+const unitPrefixes = getUnitPrefixes();
+const currencyUnits = getCurrencyUnits();
+const mathjsUnits = getAllMathjsUnits();
+
+const isValidUnit = (unit: string): boolean => {
```

Comment:
Why are we declaring all this stuff outside the describe() functions, instead of e.g. a beforeAll()? Do we follow that pattern in other test files (for anything besides simple constants)?

Reply:
I haven't found any such test where we are like creating a lot of functions, but it's still better to call these functions in beforeAll(). as other tests follow that pattern.